Compiler/Template/CodegenC.tpl

```diff
@@ -4842,14 +4814,27 @@ template equation_(Integer clockIndex, SimEqSystem eq, Context context, Text &eq
     TRACE_POP
   }
   >>
+  )
+end equation_impl;
 
+template equation_call(SimEqSystem eq, String modelNamePrefix)
+ "Generates an equation.
+  This template should not be used for a SES_RESIDUAL.
+  Residual equations are handled differently."
+::=
+  match eq
+  case e as SES_ALGORITHM(statements={})
+  then ""
+  else
+  (
+  let ix = equationIndex(eq)
```

Comment:
@sjoelund This line must be replaced to handle dynamic tearing correctly:
```
  let ix = match eq
  case e as SES_LINEAR(alternativeTearing = SOME(LINEARSYSTEM))
  case e as SES_NONLINEAR(alternativeTearing = SOME(NONLINEARSYSTEM)) then
    equationIndexAlternativeTearing(eq)
  else
    equationIndex(eq)
```

together with proposed change in line 3830 the dynamicTearing1 test runs fine again

Here, dynamicTearing2 still fails, but that one also fails for a pristine check out (stupid OS X?) :-(